src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -80,6 +101,32 @@ def spark_version_compatibility(spark_version: str) -> str:
     return "supported"
 
 
+class AzureServicePrincipalCrawler(CrawlerBase):
+    def __init__(self, ws: WorkspaceClient, sbe: SqlBackend, schema):
+        super().__init__(sbe, "hive_metastore", schema, "azure_service_principals")
+        self._ws = ws
+
+    def _crawl(self) -> list[AzureServicePrincipalInfo]:
+        relevant_service_principals = [
```

Comment:
This is the wrong way to get a list of service principals used for storage access. The logic has to be the following:
- list all clusters, collect their spark_conf
- list all jobs/tasks/job clusters, collect their spark_conf
- list all pipelines, collect their spark_conf
- list all cluster policies, collect their spark_conf.* keys
- list all sql warehouses, get their conf
- get workspace global warehouse conf
- get the relevant confs for spns
- deduplicate spns
- verify their secret is readable from a linked secret scope 
- store records in the database

Reply:
Hi @nfx - I did the first one `list all clusters, collect their spark_conf` and let me know if you are ok with this approach. I will follow teh same pattern for the rest. Thanks!